pkg/controller/operconfig/migration.go

```diff
@@ -0,0 +1,96 @@
+package operconfig
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	operv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-network-operator/pkg/apply"
+	cnoclient "github.com/openshift/cluster-network-operator/pkg/client"
+)
+
+var gvrEgressFirewall = schema.GroupVersionResource{Group: "k8s.ovn.org", Version: "v1", Resource: "egressfirewalls"}
+var gvrEgressNetworkPolicy = schema.GroupVersionResource{Group: "network.openshift.io", Version: "v1", Resource: "egressnetworkpolicies"}
+
+func migrateEgressFirewallCRs(ctx context.Context, operConfig *operv1.Network, client cnoclient.Client) error {
+	switch operConfig.Spec.DefaultNetwork.Type {
```

Comment:
Since we only check that `Migration.NetworkType` is set and not it's value we will always render those resources, even if the `Migration.NetworkType` is bogus. Understandably it is an user error but I think it would be easy to switch over `Migration.NetworkType` and not `DefaultNetwork.Type`
What do you think?

Reply:
I agree.

Reply:
Update: I'll address these changes in my own PR since they aren't critical. For now, let's get this merged.